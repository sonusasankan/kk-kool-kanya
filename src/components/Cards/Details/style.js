import styled from 'styled-components';

import Color from '../../../themes/color';


export const DetailsWrapper = styled.div`
    display: flex;
    margin-bottom: 8px;
    align-items: center;
    // justify-content: ${props => (props.center? "center": "left" )}
`;

export const Details = styled.div`
    color: ${Color.fontLight};
    font-size: 0.875rem;
    text-align: ${props => (props.center? "center": "left" )}
`;

export const Icon = styled.span`
    color: ${Color.fontLight};
    margin-right: 4px;
`;
