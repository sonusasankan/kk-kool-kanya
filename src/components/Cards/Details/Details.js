import React from 'react';
import PropTypes from 'prop-types';

import { DetailsWrapper, Details, Icon } from './style';

const details = ({
  icon,
  children,

}) => (
  <DetailsWrapper  center className="d-flex align-items-center kk-card-details__wrapper">
    {icon ? <Icon style={{"marginRight": 10}} className={`${icon} card__details__icon`} /> : ''}
    <Details className="kk-card-details">{children}</Details>
  </DetailsWrapper>
);

details.propTypes = {
  icon: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
  children: PropTypes.oneOfType([PropTypes.func, PropTypes.string]).isRequired,
  center: PropTypes.bool
};

details.defaultProps = {
  icon: false,
  center: false
};


export default details;
