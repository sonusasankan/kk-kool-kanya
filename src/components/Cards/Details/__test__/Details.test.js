import React from 'react';

import { shallow } from 'enzyme';
import { default as CardDetails } from '../Details';


describe('Details', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(
      <CardDetails icon="icon-ic-location-24">
            Bengaluru, Delhi, Hyderabad
      </CardDetails>,
    );
    expect(component).toMatchSnapshot();
  });
});
