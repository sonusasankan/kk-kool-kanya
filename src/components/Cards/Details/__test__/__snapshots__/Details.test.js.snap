// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details should render correctly in "debug" mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <details
    icon="icon-ic-location-24"
  >
    Bengaluru, Delhi, Hyderabad
  </details>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef
          className="icon-ic-location-24 mr-2 card__details__icon"
        />,
        <ForwardRef
          className="card-details"
        >
          Bengaluru, Delhi, Hyderabad
        </ForwardRef>,
      ],
      "className": "d-flex mb-2 align-items-center",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "icon-ic-location-24 mr-2 card__details__icon",
        },
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": true,
            "rules": Array [
              "
    color: ",
              "#8d959d",
              ";
    margin-right: 4px;
",
            ],
          },
          "displayName": "styled.span",
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": "span",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Bengaluru, Delhi, Hyderabad",
          "className": "card-details",
        },
        "ref": null,
        "rendered": "Bengaluru, Delhi, Hyderabad",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "rules": Array [
              "
    color: ",
              "#8d959d",
              ";
    font-size: 0.875rem;
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
    display: flex;
    margin-bottom: 8px;
    align-items: center;
",
        ],
      },
      "displayName": "styled.div",
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef
            className="icon-ic-location-24 mr-2 card__details__icon"
          />,
          <ForwardRef
            className="card-details"
          >
            Bengaluru, Delhi, Hyderabad
          </ForwardRef>,
        ],
        "className": "d-flex mb-2 align-items-center",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "icon-ic-location-24 mr-2 card__details__icon",
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "sc-htpNat",
              "isStatic": true,
              "rules": Array [
                "
    color: ",
                "#8d959d",
                ";
    margin-right: 4px;
",
              ],
            },
            "displayName": "styled.span",
            "render": [Function],
            "styledComponentId": "sc-htpNat",
            "target": "span",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Bengaluru, Delhi, Hyderabad",
            "className": "card-details",
          },
          "ref": null,
          "rendered": "Bengaluru, Delhi, Hyderabad",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": true,
              "rules": Array [
                "
    color: ",
                "#8d959d",
                ";
    font-size: 0.875rem;
",
              ],
            },
            "displayName": "styled.div",
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
    display: flex;
    margin-bottom: 8px;
    align-items: center;
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
